objects.py

```diff
@@ -14,31 +13,45 @@ def __init__():
 
         # This should set the object's price and calories, accessible by
         # .price and .calories respectively.
+        """ For creating desserts """
+        self.price = price
+        self.calories = calories
         pass
 
     # Add a calories_per_dollar method that returns the calories per dollar
     # for the dessert.
+    def calories_per_dollar(self):
+        if self.calories != None:
+            cal_per_dollar = self.calories / self.price
+            return cal_per_dollar
+        else:
+            return None
 
     # Define a method is_a_cake on Dessert that returns False
+    def is_a_cake(self):
+        return False
 
 
 class Cake(Dessert):
-
-    def __init__():
+    price = 5
+    calories = 200
+    def __init__(self, kind):
         # Edit me!
         # Cakes all cost the same amount and have the same calories, so their
         # price and calories can be set at the class-level, not during init.
         # However, we need to be able to tell cakes apart. Accept argument:
         # kind - required
-
+        self.kind = kind
         pass
 
     # Define a method is_a_cake on Cake that returns True
     # (This will override the one on Dessert)
+    def is_a_cake(self):
+        if self.price == 5 and self.calories == 200:
```

Comment:
A cake isn't defined by its price and calories, but at the object level, so if `is_a_cake()` is called on a `Cake` then this bit of code gets run -- so we just needed this to `return True` instead of checking anything more.